### ISSUE TITLE
Minimum Input support

### ISSUE THREAD

**Issue by philhrmweb** [angular-ui/ui-select#274]
There does not seem to be support for a minimum input length prior to searching? (Same as select2's Minimum Input)

Particularly in large data sets it means searching is rather slow, and 1-2 character search terms are not particularly useful.

**Comment by maxisam** [comment]
+1

**Comment by jesselpalmer** [comment]
+1

**Comment by simkessy** [comment]
Was this missed in the milestone. We're on 11.1 now but the milestone was set to 10 :/

**Comment by reichert621** [comment]
Using angular's `limitTo` in the repeater would probably help:
```<ui-select-choices repeat="item in items | filter: $select.search | limitTo: 50">```

Fixed the slowness for me, at least :grinning:

**Comment by simkessy** [comment]
There's no way it's that simple, I refuse to believe you sir! Good day!

**Comment by thefewminutes** [comment]
+1 minimum input support

**Comment by thefewminutes** [comment]
line 313 in select.js:
if (!ctrl.disabled && !ctrl.open && ctrl.search.length > 2)
also line 1359:
if ($select.tagging.isActivated && $select.search.length > 2)

**Comment by BunsenMcDubbs** [comment]
+1

**Comment by windmaomao** [comment]
looking forward to 0.13.3 for this commit, hopefully we can make this length as an option.

**Comment by windmaomao** [comment]
this commit is a bit too much, on dist ? and also it only changes two lines, but the file look like it changed a lot of places.

**Comment by windmaomao** [comment]
ok, the above commit does not work at all, and i find a easy way using just css, here's the solution.

```
                            <ui-select ng-model="selectedItem">
                                <ui-select-match>...</ui-select-match>
                                <ui-select-choices repeat="item in (itemArray | filter: $select.search | limitTo:10) track by item.id" group-by="'type'" ng-class="{hide: $select.search.length < 3 }">
                                    ...
                                </ui-select-choices>
                            </ui-select>

```

**Comment by abruere** [comment]
Sorry, I did not intend to reference this issue from my fork (lot of useless changes indeed, due to my text editor removing trailing spaces).

Thank you @thefewminutes anyway for your idea !

+1 optional minimum length support

**Comment by user378230** [comment]
Implemented in d89a869d53b918724d7a8a4c6cbcc90c477e21ea (v0.14.0)
